### ISSUE TITLE
Compatibility with Nextcloud 14

### ISSUE THREAD

**Issue by juliusknorr** [marius-wieschollek/passwords#62]
Hi,

as you may have already seen, we have entered the beta phase for the release of Nextcloud 14. We try to keep up compatibility with older apps, but in some cases apps need some adjustments to properly work on new Nextcloud releases. With Nextcloud 14 there have been quite some changes in that regard:

You can find a list of all critical changes for app developers and admins at https://github.com/nextcloud/server/issues/7827 You can run `occ app:check-code` to validate that there are no issues with deprecated or private message calls. 

I've found that you already recognized the changes in the layout at https://git.mdns.eu/nextcloud/passwords/issues/19, so https://github.com/nextcloud/server/pull/9982 should give you some hints on what you need to change.

We would be very pleased if you can make your app compatible with Nextcloud 14 and release a new version to the app store, so that users can properly test it and the upgrade experience to Nextcloud 14 will be even better than before.

Feel free to ask, if you have any questions regarding making your app compatible with Nextcloud 14. 

Btw. you might have seen that starting August 25, the Nextcloud Conference takes place at the familiar TU Berlin. (https://nextcloud.com/conf/)! We'd like all contributors to be there and that most certainly includes you. You might already have registered but if you haven't - please consider joining us!

Thanks for being part of the Nextcloud community and for all the effort you put into providing this app :rocket:

**Comment by marius-wieschollek** [comment]
Thanks for this information. I will start working on the Nextcloud 14 implementation when i'm back from my vacation.

**Comment by marius-wieschollek** [comment]
@juliushaertl When checking my app with `./occ app:check-code` i always get an error like `Syntax error, unexpected '?', expecting T_VARIABLE on line 97`. I [while back](https://help.nextcloud.com/t/occ-app-check-code-fails-with-php-7-1/25839/2) i already faced this issue and concluded that the used PhpParser version is outdated and does not support PHP 7.1 syntax.
Is this still the case or am i facing another issue here?

**Comment by juliusknorr** [comment]
As Nextcloud 14 has PHP 7.0 as its lowest supported version, we actually test with that to make sure it is still compatible. If you actually want to support just 7.1 and above you can obmit the syntax check and just go with `occ app:check-code APPID -c deprecation` to check for the APIs.

**Comment by marius-wieschollek** [comment]
@juliushaertl I tried the command, but it still does not work. I think the simple reason for that is that Nextcloud uses PhpParser 1.4.1 which was released before PHP 7.0 and definitely can't read PHP 7.1 syntax like `function test(): ?string`. 
I "upgraded" it to the latest PhpParser 4.0.4, which resolved the original error, but then crashed due to api incompatibilities.

Is it ok for you if i open an issue to upgrade this for Nextcloud 15?
